astropy/wcs/tests/test_wcs.py

```diff
@@ -864,3 +864,15 @@ def test_sip_broken():
     hdr = get_pkg_data_contents("data/sip-broken.hdr")
 
     w = wcs.WCS(hdr)
+
+
+def test_passing_ImageHDU():
+    """
+    Passing ImageHDU or PrimaryHDU directly and make sure it runs
+    """
+    paths = ['data/sip2.fits','data/validate.fits']
+    for path in paths:
+        path = get_pkg_data_filename(path)
+        hdulist = fits.open(path)
+        w=wcs.WCS(hdulist[0])
+        w.printwcs()
```

Comment:
These are essentially two identical tests. I would suggest using only one file, e.g. extensions 0 and 1 of  `validate.fits` and not using `sip2.fits`. Also would it make sense not to print the wcs but to compare two objects - one initialized with an HDU and the other initialized by passing the corresponding header? For example something like:
```
# these three should be the same
w = WCS(f[0])
w1 = WCS(f[0].header)
w = WCS()
assert w.wcs.compare(w1)
```